### PR TITLE
[NFC][SYCL] Rename poorly named diagnostic

### PR DESCRIPTION
**Pull request by Fznamznon**

### CODE REVIEW

**Review by elizabethandrews — DISMISSED** [review]

File: clang/lib/Sema/SemaDeclAttr.cpp [hunk 1]
```diff
@@ -4832,8 +4832,7 @@ static void handleSYCLDeviceIndirectlyCallableAttr(Sema &S, Decl *D,
 static void handleSYCLRegisterNumAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   auto *VD = cast<VarDecl>(D);
   if (!VD->hasGlobalStorage()) {
-    S.Diag(AL.getLoc(), diag::err_sycl_attibute_cannot_be_applied_here)
-        << AL << 0;
+    S.Diag(AL.getLoc(), diag::err_sycl_attribute_internal_function) << AL << 0;
```
Comment (elizabethandrews) [inline clang/lib/Sema/SemaDeclAttr.cpp:4835]: Actually, I just took a closer look. Why is a diagnostic about static function/function in anonymous namespace being generated for a variable. This is out of scope of your patch but this looks like incorrect diagnostic?

Comment (Fznamznon) [reply clang/lib/Sema/SemaDeclAttr.cpp:4835]: Looks like it was added by @kbobrovs in scope of https://github.com/intel/llvm/pull/1756 . Usage of this diagnostic seems a bit odd and untested.

Comment (Fznamznon) [reply clang/lib/Sema/SemaDeclAttr.cpp:4835]: Do I have to fix it?

Comment (AaronBallman) [reply clang/lib/Sema/SemaDeclAttr.cpp:4835]: If we'd like to upstream at some point, I'd recommend fixing it. (I think it's in scope for this patch given that the diagnostic quality is being regressed, FWIW).

Comment (Fznamznon) [reply clang/lib/Sema/SemaDeclAttr.cpp]: Ok. `register_num` has only 'GlobalVar' in subjects list, so it seems the part of the code where this diagnostic was incorrectly used is dead code. Removed.

**Review by elizabethandrews — APPROVED** [review]
Thanks!

File: clang/lib/Sema/SemaDeclAttr.cpp (hunk 1 above)
Comment (keryell) [reply clang/lib/Sema/SemaDeclAttr.cpp]: > If we'd like to upstream at some point

Go for it! :-)

**Pull request merged**